pkg/acsengine/engine_test.go

```diff
@@ -591,3 +591,18 @@ func TestGenerateKubeConfig(t *testing.T) {
 		t.Fatalf("Expected an error result from nil Properties child properties")
 	}
 }
+
+func TestGenerateClusterID(t *testing.T) {
+	p := &api.Properties{
+		AgentPoolProfiles: []*api.AgentPoolProfile{
+			{
+				Name: "foo_agent",
+			},
+		},
+	}
+
+	clusterID := GenerateClusterID(p)
+	if clusterID == "" {
```

Comment:
Are there any additional checks we could do like the length of the string or its content with a regexp?

Reply:
Sure. Makes sense. I can do that.